### ISSUE TITLE
Process forms in chunks or use pagination

### ISSUE THREAD

**Issue by susnux** [nextcloud/forms#2020]
This is more of a meta issue.

Currently for a lot of operations we query all (forms, submissions...) but for large instances this might get a problem if there are e.g. 30'000 forms or 3'000 submissions.

We should figure out a good way to handle:
* [ ] Show shared forms (fetching forms that are shared to the user)
* [ ] Show form results (fetching the results might be bad if there are hundreds or thousands of submissions)
* [ ] Investigate if there are more places that are problematic

**Comment by Chartman123** [comment]
findAll() in FormsMapper is only used for shared forms. So I think that we could already filter the following forms:

- `showToAllUsers` in `access_json` is `true` in forms table
- rows in shares table for `form_id` and `share_with` = current user

**Comment by Koc** [comment]
Hey mates! In theory I can work on this issue in the near future. What approach do you like:

* real pagination
  * this requires adding search, because `ctrl+f` on a page will stop works after that
* keep backend api as is, add vue-virtual-scroller
?

**Comment by Chartman123** [comment]
@susnux can we close this issue now that @Koc has implemented the pagination for submissions? I mean the summary still needs all submissions to be transmitted, because otherwise the summary stats won't be "complete". We could also only show the summary, if there are less then x submissions.

**Comment by Koc** [comment]
Yeah, I guess we need to prevent browser overload/freeze for the results summary page as well
